src/test/feature/lib/file_replace.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <fstream>
+
+#include "file_replace.h"
+
+using std::string;
+using std::map;
+using std::endl;
+using std::ifstream;
+using std::ofstream;
+
+namespace hawq {
+namespace test {
+
+string FileReplace::replaceAllOccurrences(
+	string str,
+	const string& src,
+	const string& dst)
+{
+	size_t start_pos = 0;
+	while ((start_pos = str.find(src, start_pos)) != string::npos)
+	{
+		str.replace(start_pos, src.length(), dst);
+		start_pos += dst.length();
+	}
+
+	return str;
+}
+
+void FileReplace::replace(
+	const string& file_src,
+	const string& file_dst,
+	const map<string, string>& strs_src_dst)
+{
+	ifstream fin(file_src);
+	ofstream fout(file_dst);
+	string line;
+
+	while ( getline(fin, line) )
+	{
+		typename map<string, string>::const_iterator mit;
```

Comment:
Here you don't actually need declare it, you can use `auto`: `for(auto mit = strs_src_dst.begin(); mit != strs_src_dst.end(); mit++)`. Instead, here you can use syntax like:
`for(auto & kv : mit) { auto a = kv.first; auto b = kv.second; }`